File: mobile/lib/models/person.dart
class Person {
  final int? id;
  final String firstName;
  final String? lastName;
  final String? gender;
  final String? birthDate;
  final String? deathDate;
  final String? bio;
  final String? photoUrl;
  final int? fatherId;
  final int? motherId;

  Person({
    this.id,
    required this.firstName,
    this.lastName,
    this.gender,
    this.birthDate,
    this.deathDate,
    this.bio,
    this.photoUrl,
    this.fatherId,
    this.motherId,
  });

  factory Person.fromJson(Map<String, dynamic> json) {
    return Person(
      id: json['id'],
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'],
      gender: json['gender'],
      birthDate: json['birthDate'],
      deathDate: json['deathDate'],
      bio: json['bio'],
      photoUrl: json['photoUrl'],
      fatherId: json['fatherId'],
      motherId: json['motherId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'gender': gender,
      'birthDate': birthDate,
      'deathDate': deathDate,
      'bio': bio,
      'photoUrl': photoUrl,
      'fatherId': fatherId,
      'motherId': motherId,
    };
  }

  String get fullName =>
      lastName != null && lastName!.isNotEmpty ? '$firstName $lastName' : firstName;
}

File: backend/src/main/java/com/familytree/app/person/Person.java
package com.familytree.app.person;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "persons")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    private String lastName;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    private LocalDate birthDate;

    private LocalDate deathDate;

    @Column(length = 1000)
    private String bio;

    private String photoUrl;

    // Self-referencing links to parents. Kept simple: a person can have a father and a mother.
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "father_id")
    private Person father;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mother_id")
    private Person mother;

    public enum Gender { MALE, FEMALE, OTHER, UNKNOWN }
}
